src/Mod/RuleSoldier.cpp

```diff
@@ -247,6 +211,17 @@ void RuleSoldier::afterLoad(const Mod* mod)
 			}
 		}
 	}
+	if (_skillNames.size() > 0)
+	{
+		for (auto skillName : _skillNames)
+		{
+			auto skill = mod->getSkill(skillName, true);
+			if (skill)
+			{
+				_skills.push_back(skill);
+			}
+		}
+	}
```

Comment:
`_skillNames` is not needed after this, we can clean it up. (other function doing some thing like that)